### DISCUSSION TITLE
I want to exclude requests with a status code of 4XX when calculating service_sla

### DISCUSSION THREAD

**Discussion by wangpf09** [apache/skywalking#9256] [Q&A]
I want to exclude requests with a status code of 4XX when calculating service_sla, I changed the OAL and upgraded the skywalking version, but none of them took effect
```
service_sla = from(Service.*).filter(httpResponseStatusCode != 404).percent(status == true);
```
I tried versions 8.9.1 and 8.7.0-es7 but still count 4XX requests when calculating sla;
and I found those issue `https://github.com/apache/skywalking/pull/7111` `https://github.com/apache/skywalking/issues/7135` about this question

**Comment by wu-sheng** [comment]
I am afraid you have to debug the codes on your own, such as whether your request is really carrying `httpResponseStatusCode==404`?

**Reply by wangpf09** [reply]
I just retried and confirmed a request, the status code on the page is indeed 404, but the service_sla does not filter out, Below is a screenshot of the request in skywalking

![c5def8e33f7e1035c7cbda1622890ed](https://user-images.githubusercontent.com/39894166/174926805-093f303e-2977-448b-b584-7be5151820d8.png)
![998e8cdb13dd629f2ddc2078d9394a3](https://user-images.githubusercontent.com/39894166/174926800-0fd89c6c-529c-4b3f-b5e5-b883b7b66ab0.png)
![98402323edeaf805e0cf70f2a1b9031](https://user-images.githubusercontent.com/39894166/174926804-0c2666c6-df32-48ae-82e4-31615a047277.png)

**Reply by wu-sheng** [reply]
This is not how open source works. We don't debug for your screenshots. You have the codes, you should debug them.

**Reply by wangpf09** [reply]
Ok thanks, I try to debug the code and find the problem
